server/models/procedures/stock.sql

```diff
@@ -344,3 +344,196 @@ BEGIN
+    JOIN inventory i ON i.uuid = l.inventory_uuid
+    JOIN inventory_unit iu ON iu.id = i.unit_id
+    JOIN depot d ON d.uuid = m.depot_uuid
+    LEFT JOIN document_map dm ON dm.uuid = m.document_uuid
+    WHERE m.depot_uuid = _depot_uuid AND DATE(m.date) <= _dateTo
+    ORDER BY i.text, m.date ASC;
+      
+  DECLARE CONTINUE HANDLER FOR NOT FOUND SET done = TRUE;
+
+  DROP TEMPORARY TABLE IF EXISTS stage_movement;
+  CREATE TEMPORARY TABLE stage_movement(
+    inventory_uuid BINARY(16),
+    isExit TINYINT(1),
+    qtt DECIMAL(19, 4),
+    unit_cost DECIMAL(19, 4),
+    VALUE DECIMAL(19, 4),
+    DATE DATETIME,
+    reference VARCHAR(100),
+    stockQtt DECIMAL(19, 4),
+    stockUnitCost DECIMAL(19, 4),
+    stockValue DECIMAL(19, 4),
+    iteration INT
+  );
+
+
+  OPEN curs1;
+    read_loop: LOOP
+    
+    SET mvtIsExit = 0;
+    SET mvtQtt = 0;
+    SET mvtUnitCost = 0;
+    SET mvtValue = 0;
+    SET newQuantity = 0;
+    SET newValue = 0;
+    SET newCost = 0;
+    
+    FETCH curs1 INTO _inventory_uuid, mvtIsExit, mvtUnitCost, mvtQtt, _date, _documentReference;
+      IF done THEN
+        LEAVE read_loop;
+      END IF;
+      
+      SELECT COUNT(inventory_uuid) INTO _newStock FROM stage_movement WHERE inventory_uuid = _inventory_uuid;
+      -- set stock qtt, value and unit cost for a new inventory
+      IF _newStock = 0 THEN 
+        SET stockQtt= 0;
+        SET stockUnitCost = 0;
+        SET stockValue = 0;
+        SET _iteration = 0; 
+      END IF;
+
+      -- stock exit movement, the stock quantity decreases
+      IF mvtIsExit = 1 THEN
+        SET stockQtt = stockQtt - mvtQtt;
+        SET stockValue = stockQtt * stockUnitCost;
+      ELSE
+       -- stock exit movement, the stock quantity increases
+	      SET newQuantity = mvtQtt + stockQtt;
+        SET newValue = (mvtUnitCost * mvtQtt) + stockValue;
+        SET newCost = newValue / IF(newQuantity = 0, 1, newQuantity);
+
+        SET stockQtt = newQuantity;
+        SET stockUnitCost = newCost;
+        SET stockValue = newValue;         
+      END IF;
+       
+      INSERT INTO stage_movement VALUES(
+        _inventory_uuid, mvtIsExit, mvtQtt, stockQtt, mvtQtt*mvtUnitCost, _date, _documentReference,  stockQtt, stockUnitCost, stockValue, _iteration
+      );
+      SET _iteration = _iteration + 1;
+    END LOOP;
+  CLOSE curs1;
+
+  DROP TEMPORARY TABLE IF EXISTS stage_movement_copy;
+  CREATE TEMPORARY TABLE stage_movement_copy AS SELECT * FROM stage_movement;
+
+  SELECT  BUID(sm.inventory_uuid) AS inventory_uuid, i.text as inventory_name,  sm.stockQtt, sm.stockUnitCost, sm.stockValue
+  FROM stage_movement sm
+  JOIN inventory i ON i.uuid = sm.inventory_uuid
+  INNER JOIN (
+    SELECT inventory_uuid, MAX(iteration) as max_iteration
+    FROM stage_movement_copy
+    GROUP BY inventory_uuid
+  )x ON x.inventory_uuid = sm.inventory_uuid AND x.max_iteration = sm.iteration
+
+  ORDER BY i.text ASC;
+
+  SELECT SUM(stockValue) as total FROM stage_movement_copy;
```

Comment:
Could you verify this query because the total is not equal to what we have in the table ? To verify that you can use Excel.

In Bhima : 
![image](https://user-images.githubusercontent.com/5445251/46470496-b6de1c00-c7ce-11e8-9220-fd370961071d.png)

In Excel : 
![image](https://user-images.githubusercontent.com/5445251/46470506-c65d6500-c7ce-11e8-9bb9-83d87dbea87b.png)